### ISSUE TITLE
Swallowing errors on Rails 7.1

### ISSUE THREAD

**Issue by morgoth** [rollbar/rollbar-gem#1122]
I'm integrating rollbar into Rails 7.1 Rails app (not released yet, from main branch).

In development when I enter unknown url, I don't get usual Rails informative error, but the default 500.html page and no error in the server logs.

I see that when I also enter unknown url on the production, there is no 404 page, but the 500 page and no error reported in logs nor by rollbar.

When I remove rollbar gem, then all is working as expected, so there must be some issue how the rollbar gem is intercepting request errors.

**Comment by waltjones** [comment]
@morgoth thank you for the report. I'll try it out and update here.

**Comment by morgoth** [comment]
When working on solving this issue, please consider integrating https://github.com/rollbar/rollbar-gem/issues/1105 as it might be much better and easier way to integrate with rollbar

**Comment by rgarver** [comment]
I'm not sure this is the problem but it sounds suspiciously similar. Rollbar reports nothing, but NewRelic catches the following backtrace:

```
wrong number of arguments (given 3, expected 2)
…/3.2.0/gems/rollbar-3.4.0/lib/rollbar/middleware/rails/show_exceptions.rb:7:in `render_exception_with_rollbar'
…5d93c7fa1a0/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb:44:in `rescue in call'
…5d93c7fa1a0/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
…/3.2.0/gems/rollbar-3.4.0/lib/rollbar/middleware/rails/show_exceptions.rb:22:in `call_with_rollbar'
…relic_rpm-9.3.1/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
…05d93c7fa1a0/actionpack/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
…relic_rpm-9.3.1/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
…/ruby/gems/3.2.0/gems/lograge-0.13.0/lib/lograge/rails_ext/rack/logger.rb:18:in `call_app'
…s/3.2.0/bundler/gems/rails-05d93c7fa1a0/railties/lib/rails/rack/logger.rb:24:in `block in call'
…ems/rails-05d93c7fa1a0/activesupport/lib/active_support/tagged_logging.rb:135:in `block in tagged'
…ems/rails-05d93c7fa1a0/activesupport/lib/active_support/tagged_logging.rb:39:in `tagged'
…ems/rails-05d93c7fa1a0/activesupport/lib/active_support/tagged_logging.rb:135:in `tagged'
…s/3.2.0/bundler/gems/rails-05d93c7fa1a0/railties/lib/rails/rack/logger.rb:24:in `call'
…relic_rpm-9.3.1/lib/new_relic/agent/instrumentation/middleware_tracing.rb:99:in `call'
```

**Comment by rgarver** [comment]
I confirmed that there is a change in Rails 7.1 that would produce this error: https://github.com/rails/rails/pull/46171/files#diff-9fc8f66ed0bf1b194984872ab84eeffd9f0035baa5de7446ba2a483a2bb42369R54

**Comment by rgarver** [comment]
If anyone wants a quick hack, I threw together this to make sure these exceptions are getting bubbled up to Rollbar. In theory, this is backward compatible to Rails 3.2 but I've only tested it on Rails 7.1.0.alpha:

config/initializers/rollbar.rb
```ruby
⋮

module RollbarDebugExceptionsPatch
  def render_exception(env, exception, wrapper = nil)
    key = 'action_dispatch.show_detailed_exceptions'

    if exception.is_a?(ActionController::RoutingError) && env[key]
      scope = extract_scope_from(env)

      Rollbar.scoped(scope) do
        report_exception_to_rollbar(env, exception)
      end
    end

    if self.class.instance_method(:render_exception_without_rollbar).arity == 2
      render_exception_without_rollbar(env, exception)
    else
      render_exception_without_rollbar(env, exception, wrapper)
    end
  end
end

Rails.application.config.after_initialize do
  ActionDispatch::DebugExceptions.prepend RollbarDebugExceptionsPatch
end
```

**Comment by morgoth** [comment]
@waltjones Did you look into this issue? Rails 7.1 is out, so I guess the problem will affect more people now

**Comment by brianr** [comment]
Hi all, we're looking at this issue this week. Thanks for the nudge.

**Comment by jetienne** [comment]
I am having the same issue...

**Comment by waltjones** [comment]
The fix is now released in v3.4.1: https://github.com/rollbar/rollbar-gem/releases/tag/v3.4.1

Apologies for the delay, and thanks everyone for the reports. @rgarver thank you for the analysis.

**Comment by showaltb** [comment]
FWIW, I am having this problem with an old Rails 4.2.11.3 app with rollbar gem 3.6.0 (latest).

404 errors are not being rendered and instead triggering a 500 error in all environments. Removing rollbar gem restores correct behavior.

The monkey patch from @rgarver is working for me.
